tests/testlib/s2n_connection_test_utils.c

```diff
@@ -211,6 +211,15 @@ int s2n_set_connection_hello_retry_flags(struct s2n_connection *conn)
     return 0;
 }
 
+int s2n_set_connection_server_hello_flags(struct s2n_connection *conn)
+{
+    conn->actual_protocol_version = S2N_TLS13;
+    conn->handshake.message_number = 5;
```

Comment:
Maybe the fact that we need a duplicate of this function with a different message_number means that the message_number shouldn't be set in a common test util method? Or should be an argument to the function?